fog/ledger/server/src/db_fetcher.rs

```diff
@@ -234,69 +225,70 @@ impl<
+                return 0;
             }
-            Ok(BlockDataResponse {
-                result,
-                latest_block,
-            }) => {
-                // Tracing
-                let tracer = tracer!();
-
-                let mut span = block_span_builder(&tracer, "poll_block", *next_block_index)
-                    .with_start_time(start_time)
-                    .start(&tracer);
-
-                span.set_attribute(TELEMETRY_BLOCK_INDEX_KEY.i64(*next_block_index as i64));
-
-                let _active = mark_span_as_active(span);
-
-                // Get the timestamp for the block.
-                let timestamp =
-                    if result.block_timestamp_result_code == TimestampResultCode::TimestampFound {
-                        result.block_timestamp
-                    } else {
-                        tracer.in_span("poll_block_timestamp", |_cx| {
-                            self.block_provider
-                                .poll_block_timestamp(*next_block_index, watcher_timeout)
-                        })
-                    };
-
-                // Add block to enclave.
-                let records = result
-                    .block_data
-                    .contents()
-                    .key_images
-                    .iter()
-                    .map(|key_image| KeyImageData {
-                        key_image: *key_image,
-                        block_index: *next_block_index,
-                        timestamp,
-                    })
-                    .collect();
+            Ok(blocks) => blocks,
+        };
 
-                tracer.in_span("add_records_to_enclave", |_cx| {
-                    self.add_records_to_enclave(*next_block_index, records);
-                });
+        let latest_block = blocks.latest_block;
 
-                *next_block_index += 1;
+        if let Some(next_block) = blocks.results.get(0).and_then(|r| r.as_ref()) {
+            let tracer = tracer!();
 
-                self.update_db_poll_shared_state(&latest_block, *next_block_index);
+            let mut span = block_span_builder(&tracer, "poll_block", *next_block_index)
+                .with_start_time(start_time)
+                .start(&tracer);
 
-                // Adding 1 as indices are 0 based, but "number of blocks" is 1 based.
-                latest_block.index + 1
-            }
+            span.set_attribute(TELEMETRY_BLOCK_INDEX_KEY.i64(*next_block_index as i64));
+
+            let _active = mark_span_as_active(span);
+
+            // Get the timestamp for the block.
+            let timestamp =
+                if next_block.block_timestamp_result_code == TimestampResultCode::TimestampFound {
+                    next_block.block_timestamp
+                } else {
+                    tracer.in_span("poll_block_timestamp", |_cx| {
+                        self.block_provider
+                            .poll_block_timestamp(*next_block_index, watcher_timeout)
```

Comment:
If the watcher is stuck (e.g. it's not watching the whole network so it can't get the timestamp), what happens here?

Reply:
From live discussion - we'll want to double check with infrastructure that there's alerting in place for when the watcher is stuck

Reply:
created https://sentz.atlassian.net/browse/SENTZ-2354